### PR TITLE
Generating randomly user passwords

### PR DESCRIPTION
**Pull request by davidferlay**
# Current

- Drupal admin and tester user credentials are arbitrary and statically defined in .env.default

# Expected

- User login links should be generated dynamically and randomly during `make info` makefile target

# Acceptance criteria

- Login links should display at the end of `make all` (which uses `make info`)
- Valid login links should display whenever user manually executes `make info` after build

# Approach explanation

- Using drush uli allows to not define any password, hence security

image

### CODE REVIEW

**Review by gaydabura — CHANGES_REQUESTED** [review]
Dont store in files

File: .env.default [hunk 1]
```diff
@@ -8,8 +8,6 @@ SITE_NAME=Example
 SITE_MAIL=admin@example.com
 ADMIN_NAME=admin
 ADMIN_MAIL=admin@example.com
-ADMIN_PW=admin
```
Comment (andypost) [inline .env.default:11]: I'd better renamed it to `ADMIN_PASSWORD` **or** allowed drush to generate random one

File: .env.default [hunk 2]
```diff
@@ -8,8 +8,7 @@ SITE_NAME=Example
...
-TESTER_PW=tester
+LOGIN_DESTINATION=admin/content
```
Comment (gaydabura) [inline .env.default:11]: Useless parameter. I dont see any reason to put it as variable

**Review by gaydabura — CHANGES_REQUESTED** [review]

File: .env.default (hunk 2 above)
Comment (davidferlay) [reply .env.default:11]: Reason is simple: in some specific context, such as multi-site using sub dir mode, drupal doesn't support `/login?destination=`

So having it as variable will make it easy to disable that redirect (by leaving it var empty) if need be

Comment (gaydabura) [reply .env.default:11]: I would rather suggest to use `--no-browser` argument. so command may be
`....drush user:login --name="$(TESTER_NAME)" --no-browser" | awk....`

**Review by gaydabura — CHANGES_REQUESTED** [review]

**Review by gaydabura — APPROVED** [review]

**Pull request merged**